driver-core/src/main/java/com/datastax/driver/core/SchemaParser.java

```diff
@@ -848,9 +855,344 @@ private boolean supportsUdfs(VersionNumber cassandraVersion) {
     protected static final String SELECT_VIEWS = "SELECT * FROM system_schema.views";
 
     private static final String TABLE_NAME = "table_name";
+    private static final String VIEW_NAME = "view_name";
+    private static final String COLUMN_NAME = "column_name";
+    private static final String INDEX_NAME = "index_name";
+    private static final String FUNCTION_NAME = "function_name";
+    private static final String ARGUMENT_TYPES = "argument_types";
+    private static final String AGGREGATE_NAME = "aggregate_name";
+    private static final String TYPE_NAME = "type_name";
+    private static final String LIMIT = " LIMIT 100";
+
+    private List<Row> fetchUDTs(
+        KeyspaceMetadata keyspace, Connection connection, ProtocolVersion protocolVersion)
+        throws ConnectionException, BusyConnectionException, InterruptedException,
+            ExecutionException {
+      String queryPrefix = SELECT_USERTYPES + whereClause(KEYSPACE, keyspace.getName(), null, null);
+      List<Row> result = new ArrayList<Row>();
+      List<Row> rs = queryAsync(queryPrefix + LIMIT, connection, protocolVersion).get().all();
+      while (!rs.isEmpty()) {
+        result.addAll(rs);
+        String lastSeen = "'" + result.get(result.size() - 1).getString(TYPE_NAME) + "'";
+        rs =
+            queryAsync(
+                    queryPrefix + " AND " + TYPE_NAME + " > " + lastSeen + LIMIT,
```

Comment:
Just a minor nit: is parameter binding not available at this stage?

```java
queryAsync(new SimpleStatement("... AND type_name > ? ...", lastSeen);
```

Reply:
queryAsync uses DefaultResultSetFuture that operates at a level of Message.Request which is much lower than SimpleStatement or any other Statement.